rules/code-quality/src/Rector/ClassMethod/DateTimeToDateTimeInterfaceRector.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\ClassMethod;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeInterface;
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Type\NullType as PHPStanNullType;
+use PHPStan\Type\ObjectType as PHPStanObjectType;
+use PHPStan\Type\UnionType as PHPStanUnionType;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Exception\ShouldNotHappenException;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\NodeTypeResolver\NodeTypeResolver;
+
+/**
+ * @see \Rector\CodeQuality\Tests\Rector\ClassMethod\DateTimeToDateTimeInterfaceRector\DateTimeToDateTimeInterfaceRectorTest
+ */
+final class DateTimeToDateTimeInterfaceRector extends AbstractRector
+{
+    private const METHODS_MAP = [
```

Comment:
What kind of map is this? Excluded or included?
More specific naming could help

Reply:
I've renamed it to `METHODS_RETURNING_CLASS_INSTANCE_MAP`.